src/main_nep/snes.cu

```diff
@@ -169,15 +169,34 @@ void SNES::create_population(Parameters& para)
 
 void SNES::regularize(Parameters& para)
 {
+  float lambda_1 = para.lambda_1;
+  float lambda_2 = para.lambda_2;
+  if (para.lambda_1 < 0.0f || para.lambda_2 < 0.0f) {
+    float auto_reg = 1.0e30f;
+    for (int p = 0; p < population_size; ++p) {
+      float temp = fitness[p + 3 * population_size] + fitness[p + 4 * population_size] +
+                   fitness[p + 5 * population_size];
```

Comment:
Does this mean that energies, forces and virials are weighted equally?

Reply:
the weights have already been included in this `fitness` array. You can have a look of the function `compute` in `fitness.cu`.